tests/security_test.go

```diff
@@ -295,3 +297,93 @@ func runTestCase(tc securityTestCase) error {
 
 	return err
 }
+
+func TestEnvBasedSecrets(t *testing.T) {
+	err := runSecretTest("env-based-secret", "", "secretKey", "set to the key 'secretKey' in secret 'env-based-secret'")
+
+	if err != nil {
+		t.Error(err.Error())
+	}
+}
+
+func TestFileBasedSecrets(t *testing.T) {
+	err := runSecretTest("file-based-secret", "/mnt/secrets", "", "/mnt/secrets from file-based-secret-volume")
+
+	if err != nil {
+		t.Error(err.Error())
+	}
+}
+
+func runSecretTest(secretName string, secretPath string, secretKey string, expectedSecret string) error {
+	spark := utils.SparkOperatorInstallation{}
+	err := spark.InstallSparkOperator()
+	defer spark.CleanUp()
+
+	if err != nil {
+		return err
+	}
+
+	client, err := utils.GetK8sClientSet()
+	if err != nil {
+		return err
+	}
+
+	secretData := make(map[string]string)
+	if secretKey != "" {
+		secretData[secretKey] = "secretValue"
+	} else {
+		secretData["secretKey"] = "secretValue"
+	}
+
+	err = utils.CreateSecret(client, secretName, spark.Namespace, secretData)
+	if err != nil {
+		return err
+	}
+
+	jobName := "mock-task-runner"
+	job := utils.SparkJob{
+		Name:     jobName,
+		Template: "spark-mock-task-runner-job.yaml",
+		Params: map[string]interface{}{
+			"args":       []string{"1", "15"},
+			"SecretName": secretName,
+			"SecretPath": secretPath,
+			"SecretKey":  secretKey,
+		},
+	}
+
+	err = spark.SubmitJob(&job)
+	if err != nil {
+		return err
+	}
+
+	err = spark.WaitUntilSucceeded(job)
+	if err != nil {
+		return err
+	}
+
+	jobDescription, err := utils.Kubectl(
+		"describe",
+		"pod",
+		"--namespace="+spark.Namespace,
+		jobName+"-driver",
```

Comment:
There is a private util function that does the same https://github.com/mesosphere/kudo-spark-operator/blob/ac69995410810bd83afcff3bf3b3f4c7b4b2021a/tests/utils/job.go#L80-L82  Consider making it public and using it here. It's unlikely, but if driver pod naming changes, it's better to have only one place to fix.

Reply:
Yeah it is better way. Made the method public.